File: BookExamples/56_SendBroadcast/app/src/main/java/com/ebookfrenzy/sendbroadcast/MyReceiver.java
package com.ebookfrenzy.sendbroadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

public class MyReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        //SLIGHT MODIFICATION FROM THE BOOK
        //I PASS TWO DIFFERENT INTENTS AND DEPENDING ON WHAT GETS PASSED THE CORRECT MESSAGE IS DISPLAYED.
        String msg;

        if(intent.getAction()=="com.ebookfrenzy.sendbroadcast1"){
            msg = "Broadcast intent detected 1" + intent.getAction() + intent.getExtras().getString("MyData1");
        }
        else {
            msg = "Broadcast intent detected 2" + intent.getAction() + intent.getExtras().getString("MyData2");
        }

        Toast.makeText(context, msg,
                Toast.LENGTH_LONG).show();
    }
}

File: class_book_examples/eel_and_post_ell/chapter62_SendBroadcast_eel/app/src/main/java/com/example/chapter62_sendbroadcast_eel/MyReceiver.kt
package com.example.chapter62_sendbroadcast_eel

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import android.widget.Toast

class MyReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val extras = intent.extras?:return
        var name = ""
        extras.getString("name").let {
            if (it != null) {
                name = it
            }
        }

        var message = "Broadcast intent detected " + intent.action + "  " + name

        //I DID THIS TO DETERMINE WHICH MESSAGE TO FIRE DEPENDING ON THE ACTION
        //BASICALLY I JUST CHANGE THE MESSAGE.
        if(intent.action == "android.intent.action.ACTION_POWER_DISCONNECTED"){
            message = "Power has been disconnected"
        }
        Toast.makeText(context, message,Toast.LENGTH_LONG).show()
    }
}